app/src/main/java/com/asfoundation/wallet/util/UserAgentInterceptor.kt

```diff
@@ -56,7 +56,14 @@ class UserAgentInterceptor(private val context: Context,
     val requestWithUserAgent = originalRequest.newBuilder()
       .header("User-Agent", userAgent)
       .build()
-    return chain.proceed(requestWithUserAgent)
+    val response = chain.proceed(requestWithUserAgent)
+
+    // Throw specific Exceptions on HTTP 204 and HTTP 205 response codes, since Retrofit can't handle them
+    // see retrofit issue: https://github.com/square/retrofit/issues/2867
+    if (response.code == 204) throw NoContentException("HTTP 204. There is no content")
```

Comment:
We can only map the 204 code to an error for the specific new request. The 204 is a success by default and is being used by other endpoints (for example /inapp/purchases/{uid}/consume), so it can't be an error for all those cases. I would suggest either adding a check to the originalRequest.url before throwing the exception, or mapping to a success with an empty body (also with the url check)